tests/unit/test_discovery_client.py

```diff
@@ -79,6 +91,81 @@ def test___discovery_service_unavailable___register_service_registration_failure
     assert ~unregistration_success_flag  # False
 
 
+def test___get_discovery_service_address___start_service_jit___returns_expected_value(
+    mocker: MockerFixture,
+    temp_discovery_key_file_path: pathlib.Path,
+    temp_registration_json_file_path: pathlib.Path,
+    discovery_service_temp_directory: pathlib.Path,
+):
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+        return_value=temp_discovery_key_file_path,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._service_already_running",
+        return_value=False,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
+        return_value=temp_registration_json_file_path,
+    )
+    mock_popen = mocker.patch("subprocess.Popen")
+
+    discovery_service_address = _get_discovery_service_address()
+
+    for call in mock_popen.call_args_list:
+        args, _ = call
+        (mock_file_path_arg,) = args
+        assert (
+            mock_file_path_arg[0]
+            == discovery_service_temp_directory
+            / json.loads(_MOCK_REGISTRATION_FILE_CONTENT)["discovery"]["path"]
+        )
+
+    assert mock_popen.call_count == 1
+    assert _TEST_SERVICE_PORT in discovery_service_address
+
+
+def test___get_discovery_service_address___key_file_not_exist___throws_timeouterror(
+    mocker: MockerFixture,
+    temp_discovery_key_file_path: pathlib.Path,
+    temp_registration_json_file_path: pathlib.Path,
+):
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+        return_value=temp_discovery_key_file_path,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._START_SERVICE_TIMEOUT", 5.0
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._open_key_file", side_effect=IOError
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
+        return_value=temp_registration_json_file_path,
+    )
+    mocker.patch("subprocess.Popen")
+
+    with pytest.raises(IOError) as exc_info:
+        _get_discovery_service_address()
+    assert exc_info.type is TimeoutError
+
+
+def test___start_discovery_service___key_file_exist_after_poll___service_start_success(
+    mocker: MockerFixture,
+    temp_discovery_key_file_path: pathlib.Path,
+    temp_registration_json_file_path: pathlib.Path,
+):
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._open_key_file",
+        side_effect=[IOError, IOError, IOError, temp_discovery_key_file_path],
+    )
+    mocker.patch("subprocess.Popen")
+
+    _start_service(temp_registration_json_file_path, temp_discovery_key_file_path)
```

Comment:
Verify that it started the service.

```suggestion
    mocker.patch(
        "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
        return_value=temp_registration_json_file_path,
    )
    mock_popen = mocker.patch("subprocess.Popen")

    _start_service(temp_registration_json_file_path, temp_discovery_key_file_path)
    
    exe_file_path = (discovery_service_temp_directory / json.loads(_MOCK_REGISTRATION_FILE_CONTENT)["discovery"]["path"])
    mock_popen.assert_called_once_with([exe_file_path], cwd=exe_file_path.parent)
```